dm/dm.go

```diff
@@ -71,13 +72,15 @@ var usage = func() {
 	fmt.Fprintln(os.Stderr, "Flags:")
 	flag.PrintDefaults()
 	fmt.Fprintln(os.Stderr)
-	os.Exit(1)
+	fmt.Fprintln(os.Stderr, "--stdin requires a tar archive of input files and a top level file name.")
+	fmt.Fprintln(os.Stderr)
+	panic("\n")
 }
 
 func getGitRegistry() *registry.GithubRegistry {
 	s := strings.Split(*template_registry, "/")
 	if len(s) < 2 {
-		log.Fatalf("invalid template registry: %s", *template_registry)
+		panic(fmt.Sprintf("invalid template registry: %s", *template_registry))
```

Comment:
log.Fatal skips running deferred methods. panic ensures that they run. We
handle the panic in the calling routine with recover, so there's no stack
trace. According to the Go blog, this pattern is idiomatic. Essentially the
equivalent of throwing an exception.

On Friday, November 27, 2015, vaikas-google <notifications@github.com>
wrote:

> In dm/dm.go
> <https://github.com/kubernetes/deployment-manager/pull/131#discussion_r46074842>
> :
>
> >  }
> >
> >  func getGitRegistry() *registry.GithubRegistry {
> >  	s := strings.Split(*template_registry, "/")
> >  	if len(s) < 2 {
> > -		log.Fatalf("invalid template registry: %s", *template_registry)
> > +		panic(fmt.Sprintf("invalid template registry: %s", *template_registry))
>
> What's wrong with log.Fatal? I think panic dumps a stacktrace and seems
> not that user friendly to dump the stack?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/deployment-manager/pull/131/files#r46074842>
> .
>


-- 
Jack Greenfield | Google